canvas/canvas_protocols/appointment_event.py

```diff
@@ -0,0 +1,104 @@
+
+        if changed_model == 'appointment' and self.field_changes.get('created'):
+            fhir = FumageHelper(self.settings)
+
+            response = fhir.read("Appointment", self.field_changes.get('external_id'))
+            if response.status_code != 200:
+                raise Exception("Failed to search Appointments")
+
+            patient_data = self.patient.patient
+
+            payload = {
+                "firstName": patient_data.get("firstName"),
+                "lastName": patient_data.get("lastName"),
+                "dob": patient_data.get("birthDate"),
+                "genderAtBirth": patient_data.get("sexAtBirth"),
+                "address": [
+                    {
+                        "addressLine1": patient_data["addresses"][0]["line"][0] if patient_data.get("addresses") else "",
+                        "addressLine2": patient_data["addresses"][0]["line"][1] if patient_data.get("addresses") and len(patient_data["addresses"][0]["line"]) > 1 else "",
+                        "city": patient_data["addresses"][0].get("city", "") if patient_data.get("addresses") else "",
+                        "state": patient_data["addresses"][0].get("stateCode", "") if patient_data.get("addresses") else "",
+                        "zip": patient_data["addresses"][0].get("postalCode", "") if patient_data.get("addresses") else "",
+                    }
+                ],
+                "contact": [
+                    {
+                        key: value for key, value in {
+                            "phone": next((telecom["value"] for telecom in patient_data.get("telecom", []) if telecom["system"] == "phone"), None),
+                            "email": next((telecom["value"] for telecom in patient_data.get("telecom", []) if telecom["system"] == "email"), None)
+                        }.items() if value is not None
+                    }
+                ],
+                "externalId": patient_data.get("key")
+            }
+
+            metriport_api_key = self.settings.get("METRIPORT_API_KEY")
+            metriport_facility_id = self.settings.get("METRIPORT_FACILITY_ID")
+
+            pd_response = send_notification(
+                f'{base_url}/medical/v1/patient?facilityId={metriport_facility_id}',
+                json.dumps(payload),
+                headers={
+                    'Content-Type': 'application/json',
+                    'x-api-key': metriport_api_key
+                })
+            result.add_narrative("Successfully sent patient data to Metriport API")
+        
+            pd_response_data = pd_response.json()
+            patient_id = pd_response_data.get('id')
+            result.add_narrative(json.dumps({
+                'patient_id': patient_id,
+            }))
+
+            if patient_id:
+                metadata = {
+                    'metadata': {
+                        'cx_id': f'{self.settings.get("METRIPORT_CX_ID")}',
+                        'canvas': "true"
```

Comment:
using this metadata flag to indicate that canvas flow should be followed in Metriport OSS

Reply:
Yah I can remove cx_id
canvas is used in the event emitting logic. Given your other comment though about making the event emitting logic generic I will move the logic to the canvas-event-listener.